src/resolver.ts

```diff
@@ -46,8 +46,14 @@ export default class Resolver {
 
   public async resolveGitRoot(doc?: Document): Promise<string | null> {
     let root: string
-    const { uri } = doc
-    if (!doc || doc.buftype != '' || doc.schema != 'file') {
+    let { uri } = doc
+    if (!doc) {
+      return null
+    }
+    // Support using `acwrite` with `BufWriteCmd` to create gitcommit, e.g. gina.vim
+    if (doc.buftype == 'acwrite') {
+      uri = (await workspace.nvim.eval(`getcwd(bufwinnr(${doc.bufnr}))`)) as string
```

Comment:
This will throw error when bufnr not shown in window, try `getcwd(bufwinnr(999))`

Reply:
@chemzqm Seems that I can use `doc.getcwd()` instead of this.

but I found a new problem, The value of `fullpath` is the project root directory, (like `~/repos/coc-git`), so the code `git.getRepositoryRoot(path.dirname(fullpath))` may not work.

Maybe I should check if fullpath is a directory